apps/prairielearn/elements/pl-file-preview/pl-file-preview.js

```diff
@@ -109,8 +113,22 @@
             .then(async (blob) => {
               const type = blob.type;
               if (type === 'text/plain') {
-                const text = await blob.text();
-                code.textContent = text;
+                if (escapedFileName.endsWith('.ipynb')) {
+
+                  // importing the notebookjs library doesn't return an object, it sets the global variable 'ns'
+                  // importing DOMPurify sets the global variable DOMPurify.
+                  Promise.all([import('marked'), import('purify'), import('notebook')]).then(async ([Marked]) => {                                    
+                     nb.markdown = Marked.marked.parse;
+                     nb.sanitizer = DOMPurify.sanitize;
+                     const notebook = nb.parse(JSON.parse(await blob.text()));
```

Comment:
I'd recommend moving the `await blob.text()` before you import the libraries to ensure that we consume the response body even if the dynamic imports fail for some reason.

Reply:
Would it work just to add it to `Promise.all()`?

Reply:
You could, but it's conceptually clearer if there's just a single `const text = await blob.text()` that's used in both branches here.